cmd/newrelic/command.go

```diff
@@ -150,30 +150,40 @@ func hasProfileWithDefaultName(profiles map[string]credentials.Profile) bool {
 	return false
 }
 
-func fetchLicenseKey(client *newrelic.NewRelic, accountID int) (string, error) {
+func fetchLicenseKey(ctx context.Context, client *newrelic.NewRelic, accountID int) (string, error) {
 	query := `query($accountId: Int!) { actor { account(id: $accountId) { licenseKey } } }`
 
 	variables := map[string]interface{}{
 		"accountId": accountID,
 	}
 
-	for i := 0; i < 3; i++ {
+	var licenseKey string
+	execLicenseKeyRequest := func() error {
```

Comment:
I tripped on the return "" below, only to realize this is a rather long inline function.
Is there a way to break this func out to its own private func, to avoid confusion when reading? just a suggestion.

Reply:
There is not, since it closes over `licenseKey` in order to extract the inner return value.

Reply:
I refactored this a bit.